samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -251,17 +266,20 @@ public TestRunner addOutputStream(InMemoryOutputDescriptor<?> streamDescriptor,
    * @throws SamzaException if Samza job fails with exception and returns UnsuccessfulFinish as the statuscode
    */
   public void run(Duration timeout) {
-    Preconditions.checkState(app != null,
-        "TestRunner should run for Low Level Task api or High Level Application Api");
+    Preconditions.checkState(app != null, "No Samza application to run");
     Preconditions.checkState(!timeout.isZero() || !timeout.isNegative(), "Timeouts should be positive");
+    // Cleaning store directories to ensure current run does not pick up state from previous run
+    deleteStoreDirectories();
     final LocalApplicationRunner runner = new LocalApplicationRunner(app, new MapConfig(configs));
     runner.run();
     boolean timedOut = !runner.waitForFinish(timeout);
     Assert.assertFalse("Timed out waiting for application to finish", timedOut);
     ApplicationStatus status = runner.status();
     if (status.getStatusCode() == ApplicationStatus.StatusCode.UnsuccessfulFinish) {
+      deleteStoreDirectories();
       throw new SamzaException(ExceptionUtils.getStackTrace(status.getThrowable()));
```

Comment:
What is this doing? We should not be manipulating exception stack traces.

Reply:
Looks like this is stringifying the original exception and using it as the message. You should use the constructor that accepts a (meaningful) message and a throwable. I.e., new SamzaException("Application did not finish running successfully.", e)

Reply:
when status is not successful, throwing an exception so Junit reports a failure

Reply:
I'm saying that you should throw a meaningful exception and not manipulate exception stack traces. See second comment above.